packages/wp-story-editor/src/api/media.js

```diff
@@ -129,6 +129,28 @@ export async function getOptimizedMediaById(config, mediaId) {
   return null;
 }
 
+/**
+ * Get the poster of a given media resource.
+ *
+ * @param {Object} config Configuration object.
+ * @param {number} mediaId Media ID.
+ * @return {Promise<import('@web-stories-wp/media').Resource>} Media resource if found, null otherwise.
+ */
+export async function getPosterMediaById(config, mediaId) {
+  const path = addQueryArgs(`${config.api.media}${mediaId}/`, {
+    context: 'edit',
+    _fields: 'meta.web_stories_poster_id',
+  });
+
+  const result = await apiFetch({ path });
+
+  if (result?.meta?.web_stories_poster_id) {
+    return getMediaById(config, result.meta.web_stories_poster_id);
+  }
```

Comment:
What if the user chose a custom poster image for the video? Shouldn't we use the `featured_media` field to get the poster ID?

Reply:
Good catch! Fixed in https://github.com/google/web-stories-wp/pull/9961/commits/d2781d000e567ce40d5b61a1918b776726c614a4